### PR TITLE
Removed static compilation script and unused related functions

### PR DESCRIPTION
**Pull request by HannanNaeem**
In discussion with @gliga  we decided to get rid of the legacy option to compile everything statically. This PR removes references to the particular script in question `pkc.py`

• Removed references to `pkc.py`
• Removed references to` entity_path` and `entity_name` (used for pkc) parameters in `ModuleSetup` class
• Unused functions in compiler.py using the aforementioned parameters have also been removed (`compile_sources` and `compile_entities` previously used by `pkc.py`)

### CODE REVIEW

**Review by tylerjereddy — COMMENTED** [review]
I just started the CI testing, since it doesn't run automatically for folks until their first PR gets merged. I suppose if that passes, this is probably "ok," but let's let it flush through to be safe in case you didn't test locally.

One question I have about "static compilation--" I suspect you mean removing support for ahead of time compilation rather than anything related to static linking?

**Comment by HannanNaeem** [comment]
> I just started the CI testing, since it doesn't run automatically for folks until their first PR gets merged. I suppose if that passes, this is probably "ok," but let's let it flush through to be safe in case you didn't test locally.
> 
> One question I have about "static compilation--" I suspect you mean removing support for ahead of time compilation rather than anything related to static linking?

That is indeed accurate; this is just removing support for ahead-of-time compilation and does not touch linking. Sorry for any confusion; its just the nomenclature me and Dr. Gligoric were using in our discussion. Thanks for the tests. @NaderAlAwar had some feedback I am waiting on too.

**Comment by tylerjereddy** [comment]
I haven't check the CI failure in detail yet, though it probably isn't related--could be recent releases of NumPy or `hypothesis` perhaps.

**Comment by tylerjereddy** [comment]
I'll open a separate issue for checking the failing array API CI testing, doesn't appear related.

File: pykokkos/core/module_setup.py [hunk 1]
```diff
@@ -76,22 +76,18 @@ def __init__(
...
         """
 
         self.metadata: EntityMetadata
         
-        for_pkc: bool = entity is None
-        if for_pkc:
-            self.metadata = EntityMetadata(None, entity_name, entity_path)
+        isEntityNone: bool = entity is None
+        if isEntityNone:
+            self.metadata = EntityMetadata(None, None, None)
```
[the remark this answers is not in the document]
Comment (HannanNaeem) [reply pykokkos/core/module_setup.py:90]: That is initially what I did. But I noticed this instantiation in compiler.py. I assumed the purpose of this is just to set the execution space alone? Needless to say this was throwing errors upon completely removing the if statement
`    def read_defaults(self) -> Optional[CompilationDefaults]:
        """
        Read the defaults from the defaults file

        :returns: the default values for execution space and force_uvm
        """

        defaults: Optional[CompilationDefaults]
        module_setup = ModuleSetup(None, km.get_default_space())      # <---------
        main: Path = module_setup.get_main_path()
        file: Path = self.get_defaults_file(main)

        try:
            with open(file, "r") as f:
                defaults = CompilationDefaults(**json.load(f))
        except FileNotFoundError:
            defaults = None

        return defaults`

**Pull request merged**